Expensify/App
# Setup ngrok for mWeb

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/155133

### Tests
1. First, follow the instructions in [this SO](https://stackoverflow.com/c/expensify/questions/3382) and set up a new reserved subdomain, except instead of `expensify-name` use `expensify-cash-name`.
2. Then, run `cp ./config/ngrok.example.yml ./config/ngrok.yml`
3. Replace your name and auth token in `./config/ngrok.yml`
4. Update your `.env`:
    1) Change `NGROK_URL` to `NGROK_URL_COM`
    2) Change `USE_NGROK` to `USE_NGROK_COM`
    3) Add `NGROK_URL_CASH=https://expensify-cash-$REPLACE.ngrok.io`
    4) Add `USE_NGROK_CASH=true`
    5) If it's not already true, set `USE_NGROK_COM=true`
5. Run `npm run ngrok`. Make sure it's forwarding traffic from `expensify-$REPLACE.ngrok.io` -> `expensify.com.dev` and from `expensify-cash-$REPLACE.ngrok.io` -> `localhost:8080`. I had to run it twice for it to work.
6. In another terminal tab, run `npm run web`
7. On your mobile device, go to `https://expensify-cash-$REPLACE.ngrok.io`. It will take a little while on the first load, but then should be pretty fast (since the whole app bundle will be downloaded)
8. Sign in to the app (to make sure you can connect to the expensify.com.dev API).
9. Make a noticeable change to the code and save it. Wait a minute and see that that change appears via webpack's hot-reload. It's still a bit slow, but was faster than the initial load in my experience.
10. Kill the process running webpack (the terminal tab you ran `npm run web` in)
11. Change `USE_NGROK_CASH` to `false` in your `.env`
14. Re-run `npm run web` and visit `localhost:8080` in your browser on your laptop. Make sure the app loads.
15. Make a noticeable change to the code and save it. It should hot-reload the change in your browser window. You should still be seeing your API requests go through `ngrok`

### Tested On


### Screenshots
#### Web
![image](https://user-images.githubusercontent.com/47436092/109245184-dd80c100-7794-11eb-88fe-0724af3f00c1.png)

#### Mobile Web
<img src="https://user-images.githubusercontent.com/47436092/109245292-086b1500-7795-11eb-8bb0-077402175152.jpeg" alt="" width="350px" />

#### Desktop


#### iOS


#### Android
